src/WormholeEndpoint.sol

```diff
@@ -105,23 +130,26 @@ abstract contract WormholeEndpoint is Endpoint {
         return (EndpointStructs.encodeEndpointMessage(endpointMessage), endpointMessage);
     }
 
-    function _sendMessage(uint16 recipientChain, bytes memory payload) internal override {
+    function _sendMessage(uint16 recipientChain, bytes memory managerMessage) internal override {
         (
             bytes memory encodedEndpointPayload,
             EndpointStructs.EndpointMessage memory endpointMessage
-        ) = wrapManagerMessageInEndpoint(payload);
-
-        // do not use standard relaying for solana deliveries
-        if (recipientChain == 1) {
-            wormhole().publishMessage(0, encodedEndpointPayload, 1);
+        ) = wrapManagerMessageInEndpoint(managerMessage);
+
+        if (isWormholeRelayingEnabled(recipientChain)) {
+            if (isWormholeEvmChain(recipientChain)) {
+                wormholeRelayer.sendPayloadToEvm{value: msg.value}(
+                    recipientChain,
+                    fromWormholeFormat(getWormholeSibling(recipientChain)),
+                    encodedEndpointPayload,
+                    0,
+                    GAS_LIMIT
+                );
+            } else {
+                revert RelayingNotImplemented(recipientChain);
+            }
         } else {
-            wormholeRelayer().sendPayloadToEvm{value: msg.value}(
-                recipientChain,
-                fromWormholeFormat(getWormholeSibling(recipientChain)),
-                encodedEndpointPayload,
-                0,
-                _GAS_LIMIT
-            );
+            wormhole.publishMessage(0, encodedEndpointPayload, 1);
```

Comment:
this logic is flipped? this should revert relaying is not enabled and emit via the core bridge when its not an evm chain.

Reply:
maybe better to define a predicate for when to emit via core bridge and then another predicate for emitting via AR. `relayViaAR` and `emitViaCoreBridge`